public/src/components/switchboard.tsx

```diff
@@ -97,61 +136,164 @@ const Switchboard: React.FC<InnerProps<SupportFrontendSwitches>> = ({
   const classes = useStyles();
 
   const [needToSaveDataWarning, setNeedToSaveDataWarning] = useState(false);
+  const [newUnsavedData, setNewUnsavedData] = useState(false);
+  const [pendingChanges, setPendingChanges] = useState<string[]>([]);
+
+  type FormData = {
+    switchId: string;
+    description: string;
+  };
 
   const displayNeedToSaveDataWarning = (): JSX.Element | false => {
     return (
-      needToSaveDataWarning && (
+      needToSaveDataWarning &&
+      newUnsavedData && (
         <Alert severity="warning">
           Switch settings have been changed. Changes need to be saved before they take effect!
+          <List>
+            {pendingChanges.map((change, index) => (
+              <ListItem key={index}>
+                <ListItemText primary={change} />
+              </ListItem>
+            ))}
+          </List>
         </Alert>
       )
     );
   };
 
+  const setPendingChange = (changeName: string, description: string, groupId: string): void => {
+    const unsavedChange = changeName + ' ' + description + ' in ' + groupId;
+    // Add the changeName to the list of pendingChanges
+    setPendingChanges(prevChanges => [...prevChanges, unsavedChange]);
+  };
+
   const updateSwitchSetting = (
     switchId: string,
     switchData: Switch,
-    groupId: string,
+    group: [string, SwitchGroup],
     isChecked: boolean,
   ): void => {
     const updatedState = cloneDeep(data);
-
+    const [groupId, groupData] = group;
     updatedState[groupId].switches[switchId].state = isChecked ? SwitchState.On : SwitchState.Off;
-
+    const currentSwitchState = updatedState[groupId].switches[switchId].state;
     setData(updatedState);
+    setNewUnsavedData(true);
+    setPendingChange(
+      'Turned ' + currentSwitchState + ':',
+      switchData.description,
+      groupData.description,
+    );
     setNeedToSaveDataWarning(true);
   };
 
   const createSwitchesFromGroupData = (
     switchId: string,
     switchData: Switch,
-    groupId: string,
+    group: [string, SwitchGroup],
   ): JSX.Element => {
     const isChecked = switchData.state === 'On';
 
     return (
-      <FormControlLabel
-        label={switchData.description}
-        checked={switchData.state === 'On'}
-        control={<SwitchUI />}
-        onChange={(): void => updateSwitchSetting(switchId, switchData, groupId, !isChecked)}
-        value={switchId}
-        key={switchId}
-      />
+      <div className={classes.switchLayout}>
+        <FormControlLabel
+          label={switchData.description}
+          checked={switchData.state === 'On'}
+          control={<SwitchUI />}
+          onChange={(): void => updateSwitchSetting(switchId, switchData, group, !isChecked)}
+          value={switchId}
+          key={switchId}
+        />
+        <IconButton
+          onClick={(): void => actionRemoveSwitchData(group, switchId, switchData.description)}
+          aria-label="removeSwitch"
```

Comment:
From my quick reading of [the MDN docs for aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label), it sounds like this should probably be a description a screen reader can read out, rather than an ID label. I.e. I think this would be better with spaces:

```suggestion
          aria-label="Remove switch"
```